src/pages/Swap/index.tsx

```diff
@@ -123,11 +122,11 @@ const DetailsSwapSection = styled(SwapSection)`
 `
 
 function getIsValidSwapQuote(
-  trade: InterfaceTrade<Currency, Currency, TradeType> | undefined,
+  trade: InterfaceTrade | undefined,
   tradeState: TradeState,
   swapInputError?: ReactNode
 ): boolean {
-  return !!swapInputError && !!trade && (tradeState === TradeState.VALID || tradeState === TradeState.SYNCING)
+  return !!swapInputError && !!trade && tradeState === TradeState.VALID
```

Comment:
Instead of `!!`, what do you think of `Boolean(swapInputError && trade && tradeState === TradeState.VALID)`?